### ISSUE TITLE
2.13.x: ScalaTest-related failures: akka-persistence-jdbc,scala-pe​t-store

### ISSUE THREAD

**Issue by SethTisue** [scala/community-builds#1072]
perhaps some nondeterminism lurking? since otherwise it's difficult to understand:

* why it's JDK 11 only, not 8 ~~or 14~~
* why it doesn't happen in my local runs on macOS

https://scala-ci.typesafe.com/job/scala-2.13.x-jdk11-integrate-community-build/1148/artifact/logs/akka-persistence-jdbc-build.log

```
[akka-persistence-jdbc] [error] /home/jenkins/workspace/scala-2.13.x-jdk11-integrate-community-build/target-0.9.16/project-builds/akka-persistence-jdbc-08276e0bbfd698e2f20a60cd4bc5174e81efb227/src/test/scala/akka/persistence/jdbc/query/JournalSequenceActorTest.scala:73:14: not enough arguments for method eventually: (implicit config: JournalSequenceActorTest.this.PatienceConfig, implicit retrying: org.scalatest.enablers.Retrying[org.scalatest.Assertion], implicit pos: org.scalactic.source.Position)org.scalatest.Assertion.
[akka-persistence-jdbc] [error] Unspecified value parameter pos.
[akka-persistence-jdbc] [error]             }(patienceConfig, implicitly)
[akka-persistence-jdbc] [error]              ^
```

https://scala-ci.typesafe.com/job/scala-2.13.x-jdk11-integrate-community-build/1148/artifact/logs/scala-pet-store-build.log

```
[scala-pet-store] [error] /home/jenkins/workspace/scala-2.13.x-jdk11-integrate-community-build/target-0.9.16/project-builds/scala-pet-store-504250148a48986ef4e1c4ca9590fb1846f7b49a/src/test/scala/io/github/pauljamescleary/petstore/infrastructure/endpoint/OrderEndpointsSpec.scala:48:5: Symbol 'type org.scalatest.prop.Configuration.PropertyCheckConfigurable' is missing from the classpath.
[scala-pet-store] [error] This symbol is required by 'value org.scalatestplus.scalacheck.ScalaCheckDrivenPropertyChecks.config'.
[scala-pet-store] [error] Make sure that type PropertyCheckConfigurable is in your classpath and check for conflicting dependencies with `-Ylog-classpath`.
[scala-pet-store] [error] A full rebuild may help if 'ScalaCheckDrivenPropertyChecks.class' was compiled against an incompatible version of org.scalatest.prop.Configuration.
[scala-pet-store] [error]     forAll { (order: Order, user: AdminUser) =>
[scala-pet-store] [error]     ^
```

**Comment by SethTisue** [comment]
this will be a full rebuild, because the sbt version was bumped: https://scala-ci.typesafe.com/job/scala-2.13.x-jdk11-integrate-community-build/1157/

**Comment by SethTisue** [comment]
seen on JDK 14, where we haven't seen it before, supporting the hypothesis that there's nondeterminism here: https://scala-ci.typesafe.com/job/scala-2.13.x-jdk14-integrate-community-build/668/artifact/logs/akka-persistence-jdbc-build.log

**Comment by SethTisue** [comment]
the akka-persistence-jdbc error is reproducible outside of dbuild by bumping the ScalaTest version from 3.0.8 to 3.1.0 without changing anything else, and then running `core/Test/compile`

so, as I'd already assumed, this is a ScalaTest 3.0 vs 3.1 thing

I thought perhaps both 3.0 and 3.1 were somehow ending up on the classpath. but I don't see evidence of that here:

```
[akka-persistence-jdbc] All Dependencies for subproject default-sbt-project:
[akka-persistence-jdbc]    org.scala-lang:scala-compiler:2.13.2-bin-3d4c087:scala-tool->default,optional(default)
[akka-persistence-jdbc]    org.scala-lang:scala-library:2.13.2-bin-3d4c087:scala-tool->default,optional(default)
[akka-persistence-jdbc]    com.lightbend:cloc-plugin:0.0.1-dbuildxb1b76e2e27390459097dd3891afa027f943f23af:plugin->default(compile)
[akka-persistence-jdbc]    scalacommunitybuild:scalatest:3.0.8-dbuildx0cccd5ea47ea4d2b9400b7191de74f5e75cf62b6:test
[akka-persistence-jdbc]    org.scala-lang:scala-library:2.13.2-bin-3d4c087
[akka-persistence-jdbc]    com.typesafe.akka:akka-actor:2.6.3-dbuildx2217b05c00ebbead7ebe42dd7a694d292d6b2955
[akka-persistence-jdbc]    com.typesafe.akka:akka-persistence:2.6.3-dbuildx532698f91bf920c8a8cc2630822369b5398425d3
[akka-persistence-jdbc]    com.typesafe.akka:akka-persistence-query:2.6.3-dbuildx532698f91bf920c8a8cc2630822369b5398425d3
[akka-persistence-jdbc]    com.typesafe.akka:akka-stream:2.6.3-dbuildx0a5568377fde002be84388004c63ba445079fc75
[akka-persistence-jdbc]    com.typesafe.slick:slick:3.4.0-dbuildx2babf42be14d790f70119c633afc614743d74682
[akka-persistence-jdbc]    com.typesafe.slick:slick-hikaricp:3.4.0-dbuildx2babf42be14d790f70119c633afc614743d74682
[akka-persistence-jdbc]    org.postgresql:postgresql:42.2.5:test
[akka-persistence-jdbc]    com.h2database:h2:1.4.199:test
[akka-persistence-jdbc]    mysql:mysql-connector-java:8.0.15:test
[akka-persistence-jdbc]    com.microsoft.sqlserver:mssql-jdbc:7.2.1.jre8:test
[akka-persistence-jdbc]    ch.qos.logback:logback-classic:1.2.3:test
[akka-persistence-jdbc]    com.typesafe.akka:akka-slf4j:2.6.3-dbuildx2217b05c00ebbead7ebe42dd7a694d292d6b2955:test
[akka-persistence-jdbc]    com.typesafe.akka:akka-persistence-tck:2.6.3-dbuildx532698f91bf920c8a8cc2630822369b5398425d3:test
[akka-persistence-jdbc]    com.typesafe.akka:akka-stream-testkit:2.6.3-dbuildx0a5568377fde002be84388004c63ba445079fc75:test
[akka-persistence-jdbc]    com.typesafe.akka:akka-testkit:2.6.3-dbuildx2217b05c00ebbead7ebe42dd7a694d292d6b2955:test
```

looks like just 3.0

unless somehow one of the Akka testkit jars embeds 3.1?

this isn't embedding, just a dependency, but I do see that the akka-testkit and akka-stream-testkit POMs have

```
        <dependency>
            <groupId>org.scalatest</groupId>
            <artifactId>scalatest_2.13</artifactId>
            <version>3.1.0</version>
            <scope>test</scope>
        </dependency>
```

I think the simplest way out here is if I PR a 3.0->3.1 upgrade (with an Akka 2.6.0->2.6.3 bump as well) in the akka-persistence-jdbc repo. or maybe don't PR it but just hack-and-slash it in our fork. I don't know if it's intentional that akka-persistence-jdbc has kept its Akka dependency at 2.6.0

**Comment by SethTisue** [comment]
our akka-persistence-jdbc is frozen, and the repo has different dependency stuff on master, so I think just forking but not bothering with a PR is the way to go here

**Comment by SethTisue** [comment]
okay, hopefully #1073 will take care of akka-persistence-jdbc.

now, as for scala-pet-store, hmm:

```
[scala-pet-store] All Dependencies for subproject default-sbt-project:
[scala-pet-store]    org.scala-lang:scala-compiler:2.13.2-bin-3d4c087:scala-tool->default,optional(default)
[scala-pet-store]    org.scala-lang:scala-library:2.13.2-bin-3d4c087:scala-tool->default,optional(default)
[scala-pet-store]    com.lightbend:cloc-plugin:0.0.1-dbuildx2cf0c3552a2d0493c6c89ce4941b8a289d6f7351:plugin->default(compile)
[scala-pet-store]    org.scala-lang:scala-library:2.13.2-bin-3d4c087
[scala-pet-store]    org.tpolecat:tut-core:0.6.14-dbuildx6a3100ddcbb6e12dc0872b8e018ff1c45683b9c9:Tut
[scala-pet-store]    org.scalameta:mdoc:2.0.3-dbuildx33afd4b04bcdde41c9d08f13c32feb0d011f5b24
[scala-pet-store]    org.typelevel:cats-core:2.1.1-dbuildx362114b1bc197fabcba0e939417ad5568f5b4ced
[scala-pet-store]    io.circe:circe-generic:0.13.0-dbuildxfa728cfe2d427151eb55aaff263562d845405858
[scala-pet-store]    io.circe:circe-literal:0.13.0-dbuildxfa728cfe2d427151eb55aaff263562d845405858
[scala-pet-store]    io.circe:circe-generic-extras:0.12.3-dbuildxc4a72768f94ac805234acc75c98200dd820aeadb
[scala-pet-store]    io.circe:circe-parser:0.13.0-dbuildxfa728cfe2d427151eb55aaff263562d845405858
[scala-pet-store]    io.circe:circe-config:0.7.0-dbuildxe2c82cc914c1063204688c0e2429d1cc88348fe3
[scala-pet-store]    org.tpolecat:doobie-core:0.8.8-dbuildxcd4efe91a794e63eb6fad820ce7aa27c95f4f869
[scala-pet-store]    org.tpolecat:doobie-h2:0.8.8-dbuildxcd4efe91a794e63eb6fad820ce7aa27c95f4f869
[scala-pet-store]    org.tpolecat:doobie-scalatest:0.8.8-dbuildxcd4efe91a794e63eb6fad820ce7aa27c95f4f869
[scala-pet-store]    org.tpolecat:doobie-hikari:0.8.8-dbuildxcd4efe91a794e63eb6fad820ce7aa27c95f4f869
[scala-pet-store]    com.beachape:enumeratum-circe:0.1.0-dbuildxfcd6972b40e73df458443c8df0be4a312f45f2a7
[scala-pet-store]    com.h2database:h2:1.4.200
[scala-pet-store]    org.http4s:http4s-blaze-server:0.21.0-dbuildxa485034dee03ca71e1acba128736e7ee42cfe3e3
[scala-pet-store]    org.http4s:http4s-circe:0.21.0-dbuildxa485034dee03ca71e1acba128736e7ee42cfe3e3
[scala-pet-store]    org.http4s:http4s-dsl:0.21.0-dbuildxa485034dee03ca71e1acba128736e7ee42cfe3e3
[scala-pet-store]    ch.qos.logback:logback-classic:1.2.3
[scala-pet-store]    org.flywaydb:flyway-core:6.1.4
[scala-pet-store]    org.http4s:http4s-blaze-client:0.21.0-dbuildxa485034dee03ca71e1acba128736e7ee42cfe3e3:test
[scala-pet-store]    org.scalacheck:scalacheck:0.1.0-dbuildx73e54e3c2dc1ed4b39e2589641978d10cc5344ef:test
[scala-pet-store]    org.scalatest:scalatest:0.1.0-dbuildxd4adf029774ac2b70d237015e3ea4270c040516a:test
[scala-pet-store]    org.scalatestplus:scalacheck-1-14:0.0.0-dbuildx1e40394979744692f543e18ac0b0c6c1b8978b31:test
[scala-pet-store]    io.github.jmcardon:tsec-common:0.2.0-dbuildx0c32505f761ea5dbc2356556ba41b34b21d5a0c1
[scala-pet-store]    io.github.jmcardon:tsec-password:0.2.0-dbuildx0c32505f761ea5dbc2356556ba41b34b21d5a0c1
[scala-pet-store]    io.github.jmcardon:tsec-mac:0.2.0-dbuildx0c32505f761ea5dbc2356556ba41b34b21d5a0c1
[scala-pet-store]    io.github.jmcardon:tsec-signatures:0.2.0-dbuildx0c32505f761ea5dbc2356556ba41b34b21d5a0c1
[scala-pet-store]    io.github.jmcardon:tsec-jwt-mac:0.2.0-dbuildx0c32505f761ea5dbc2356556ba41b34b21d5a0c1
[scala-pet-store]    io.github.jmcardon:tsec-jwt-sig:0.2.0-dbuildx0c32505f761ea5dbc2356556ba41b34b21d5a0c1
[scala-pet-store]    io.github.jmcardon:tsec-http4s:0.2.0-dbuildx0c32505f761ea5dbc2356556ba41b34b21d5a0c1
[scala-pet-store]    org.typelevel:kind-projector:0.11.1-dbuildx77229d107d350ff8e7e1def62d9a028428615ed0:plugin->default(compile)
```

the scala-pet-store repo itself is on ScalaTest 3.1, so I hypothesize we have the opposite problem here: somewhere, a dependency on 3.0

an obvious suspect would be doobie-scalatest. I'll pursue that hypothesis now

**Comment by SethTisue** [comment]
added doobie fix to #1073

**Comment by SethTisue** [comment]
optimistically closing on the assumption that all will be well now that #1073 is merged, knock on wood
